janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/computer/VertexMemoryHandler.java

```diff
@@ -154,9 +153,7 @@ public void sendMessage(MessageScope messageScope, M m) {
             vertexMemory.sendMessage(vertexId, m, messageScope);
         } else {
             ((MessageScope.Global) messageScope).vertices().forEach(v -> {
-                long vertexId;
-                if (v instanceof JanusGraphVertex) vertexId=((JanusGraphVertex)v).longId();
-                else vertexId = (Long)v.id();
+                long vertexId = ((Number) v.id()).longValue();
                 vertexMemory.sendMessage(vertexMemory.getCanonicalId(vertexId), m, messageScope);
```

Comment:
This PR does not intend to bring about any breaking or functional change. For now, we just assume its a long value, but in the subsequent PRs, this will be changed. The message bus will also require breaking changes.

Reply:
btw

>  You could potentially also change signature of the next method

`forceRemoveVertexFromGraphIndex` is changed in this PR. Maybe you are looking at the wrong version?

Reply:
You right. I missed that. Thank you!